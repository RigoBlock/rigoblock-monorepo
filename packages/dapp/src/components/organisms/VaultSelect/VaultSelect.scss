@import "../../_settings/variables.scss";

.vault-select {
  max-width: 4 * $dist-6xl;
  .active {
    border: solid 2px $azure-radiance;
    padding: ($dist-s - 2px);
  }
  text-decoration: none;
  .list-item {
    background: $porcelain;
    &:nth-child(2n) {
      background: $md-secondary-color;
    }
  }
}
