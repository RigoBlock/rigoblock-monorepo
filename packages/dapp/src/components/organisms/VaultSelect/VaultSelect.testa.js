import { Provider } from 'react-redux'
import { mount } from 'enzyme'
import React from 'react'
import VaultList from './VaultList.jsx'
import toJson from 'enzyme-to-json'

const mockStore = {
  getState: () => ({
    user: {
      preferences: {
        currentAccount: '0x242B2Dd21e7E1a2b2516d0A3a06b58e2D9BF9196'
      },
      blockChain: {
        accounts: {
          '0x242B2Dd21e7E1a2b2516d0A3a06b58e2D9BF9196': {
            vaults: {
              '0x86a1ba4d485ce346bded508e2426798f825558be': {
                id: 0,
                name: 'First Vault',
                symbol: 'asd',
                owner: '0x242b2dd21e7e1a2b2516d0a3a06b58e2d9bf9196',
                group: '0x7ce6e371085cb611fb46d5065397223ef2f952ff',
                totalSupply: 14
              },
              '0x421e1cef6e85e78da2470e54af64a626f45afb85': {
                id: 2,
                name: 'Third Vault',
                symbol: 'das',
                owner: '0x7328ef1d7ab7583eb9968b2f4a9c900f8a2e2d6d',
                group: '0x7ce6e371085cb611fb46d5065397223ef2f952ff',
                totalSupply: 20.23498
              }
            }
          }
        }
      }
    }
  }),
  dispatch: jest.fn(),
  subscribe: () => null
}

const noVaultsStore = {
  getState: () => ({
    user: {
      preferences: {
        currentAccount: '0x242B2Dd21e7E1a2b2516d0A3a06b58e2D9BF9196'
      },
      blockChain: {
        accounts: {
          '0x242B2Dd21e7E1a2b2516d0A3a06b58e2D9BF9196': {}
        }
      }
    }
  }),
  dispatch: jest.fn(),
  subscribe: () => null
}

const wrapper = mount(
  <Provider store={mockStore}>
    <VaultList />
  </Provider>
)

const noVaultWrapper = mount(
  <Provider store={noVaultsStore}>
    <VaultList />
  </Provider>
)

describe('PreferencesForm component', () => {
  it('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders correctly if there are no vaults on state', () => {
    expect(toJson(noVaultWrapper)).toMatchSnapshot()
  })
})
