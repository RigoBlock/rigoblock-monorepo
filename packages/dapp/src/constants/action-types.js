import keyMirror from 'keymirror'

export const actionTypes = keyMirror({
  // Global actions
  GLOBAL_INIT: null,
  // Blockchain actions
  BLOCKCHAIN_INIT: null,
  BLOCKCHAIN_ERROR: null,
  LOGGED_IN: null,
  LOGGED_OUT: null
})
