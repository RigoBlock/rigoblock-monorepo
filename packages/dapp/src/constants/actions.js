export default {
  persist: 'persist',
  form: '@@redux-form'
}
