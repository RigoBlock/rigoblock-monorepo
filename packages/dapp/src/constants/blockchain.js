export const VAULT = 'vault'
export const DRAGO = 'drago'
