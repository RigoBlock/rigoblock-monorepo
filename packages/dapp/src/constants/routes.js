export const DASHBOARD = '/'
export const PREFERENCES = '/preferences'
export const HELP = '/help'
export const LOGIN = '/login'
export const VAULTS = '/vaults'
