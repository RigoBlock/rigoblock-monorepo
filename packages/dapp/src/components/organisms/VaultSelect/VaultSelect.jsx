import './VaultSelect.scss'
import * as ROUTES from '../../../constants/routes'
import { BigNumber } from 'bignumber.js'
import { ETH_TO_WEI } from '../../../constants/utils'
import { MANAGER } from '../../../constants/user'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import Button, { BUTTON_TYPES } from '../../atoms/Button'
import CreateVaultModal from '../CreateVaultModal'
import List from '../../organisms/List'
import ListItem from '../../molecules/ListItem'
import PropTypes from 'prop-types'
import React from 'react'
import classNames from 'classnames'
import get from 'lodash/get'
import globalActions from '../../../actions/global-actions'
import routerActions from '../../../actions/router-actions'

let vaultSelect = ({
  vaults,
  location,
  userType,
  openModal,
  navigateToVault
}) => {
  // TODO: remove this and implement a correct page
  const handleVaultClick = ({ target }) => {
    return location === `${ROUTES.VAULTS}/${target.id}`
      ? null
      : navigateToVault(target.id)
  }

  const handleButtonClick = () => openModal(CreateVaultModal)

  if (!vaults) {
    return (
      <Button appearance={BUTTON_TYPES.INVERTED} onClick={handleButtonClick}>
        New Vault
      </Button>
    )
  }

  let vaultsList = Object.values(vaults).map(
    ({ name, symbol, totalSupply, id }) => ({
      itemName: name,
      itemSymbol: symbol,
      itemValue: totalSupply ? totalSupply.div(ETH_TO_WEI) : new BigNumber('0'),
      id,
      onClick: handleVaultClick
    })
  )

  vaultsList = vaultsList.map(vault => ({
    ...vault,
    className: classNames({
      active: location === `${ROUTES.VAULTS}/${vault.id.toString()}`
    })
  }))

  return userType === MANAGER ? (
    <div className="vault-select">
      <Button appearance={BUTTON_TYPES.INVERTED} onClick={handleButtonClick}>
        New Vault
      </Button>
      <List Component={ListItem} data={vaultsList} />
    </div>
  ) : (
    <div className="vault-select">
      <List Component={ListItem} data={vaultsList} />
    </div>
  )
}

vaultSelect.propTypes = {
  vaults: PropTypes.object,
  openModal: PropTypes.func.isRequired,
  navigateToVault: PropTypes.func.isRequired,
  location: PropTypes.string.isRequired,
  userType: PropTypes.string.isRequired
}

vaultSelect.defaultProps = {
  vaults: {}
}

vaultSelect = connect(
  state => {
    const { currentAccount } = state.preferences
    return {
      vaults:
        currentAccount &&
        get(state, `blockChain.accounts[${currentAccount}].vaults`, null),
      location: state.routing.location.pathname,
      userType: state.preferences.type
    }
  },
  dispatch => ({
    ...bindActionCreators(globalActions, dispatch),
    ...bindActionCreators(routerActions, dispatch)
  })
)(vaultSelect)

export default vaultSelect
