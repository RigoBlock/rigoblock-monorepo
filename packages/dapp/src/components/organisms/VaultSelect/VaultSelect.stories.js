import '../../_settings/_base.scss'
import * as ROUTES from '../../../constants/routes'
import { MemoryRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import { selectV2, withKnobs } from '@storybook/addon-knobs/react'
import { storiesOf } from '@storybook/react'
import React from 'react'
import VaultSelect from './VaultSelect'
import mockStore, { defaultState } from '../../../fixtures/store'

const locations = {
  firstVault: `${ROUTES.VAULTS}0`,
  secondVault: `${ROUTES.VAULTS}1`
}
const getState = () => ({
  ...defaultState,
  ...{
    routing: {
      location: {
        pathname: selectV2('Location', locations, locations.firstVault)
      }
    }
  }
})

const store = mockStore({ getState })

storiesOf('Organisms/VaultSelect', module)
  .addDecorator(withKnobs)
  .addDecorator(story => <MemoryRouter>{story()}</MemoryRouter>)
  .addDecorator(story => <Provider store={store}>{story()}</Provider>)
  .add('default', () => <VaultSelect />)
