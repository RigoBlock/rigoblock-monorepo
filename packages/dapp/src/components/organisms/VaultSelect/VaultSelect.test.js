import * as ROUTES from '../../../constants/routes'
import { BigNumber } from 'bignumber.js'
import { Provider } from 'react-redux'
import { mount } from 'enzyme'
import React from 'react'
import VaultSelect from './VaultSelect.jsx'
import toJson from 'enzyme-to-json'

const mockStore = {
  getState: () => ({
    preferences: {
      currentAccount: '0x242B2Dd21e7E1a2b2516d0A3a06b58e2D9BF9196'
    },
    blockChain: {
      accounts: {
        '0x242B2Dd21e7E1a2b2516d0A3a06b58e2D9BF9196': {
          vaults: {
            '0x86a1ba4d485ce346bded508e2426798f825558be': {
              id: new BigNumber(0),
              name: 'First Vault',
              symbol: 'asd',
              owner: '0x242b2dd21e7e1a2b2516d0a3a06b58e2d9bf9196',
              group: '0x7ce6e371085cb611fb46d5065397223ef2f952ff',
              totalSupply: new BigNumber(14000000)
            },
            '0x421e1cef6e85e78da2470e54af64a626f45afb85': {
              id: 2,
              name: 'Third Vault',
              symbol: 'das',
              owner: '0x7328ef1d7ab7583eb9968b2f4a9c900f8a2e2d6d',
              group: '0x7ce6e371085cb611fb46d5065397223ef2f952ff',
              totalSupply: new BigNumber(2023498)
            }
          }
        }
      }
    },
    routing: {
      location: {
        pathname: `${ROUTES.VAULTS}/0`
      }
    }
  }),
  dispatch: jest.fn(),
  subscribe: () => null
}

const noVaultsStore = {
  getState: () => ({
    preferences: {
      currentAccount: '0x242B2Dd21e7E1a2b2516d0A3a06b58e2D9BF9196'
    },
    blockChain: {
      accounts: {
        '0x242B2Dd21e7E1a2b2516d0A3a06b58e2D9BF9196': {}
      }
    },
    routing: {
      location: {
        pathname: ROUTES.VAULTS
      }
    }
  }),
  dispatch: jest.fn(),
  subscribe: () => null
}

const wrapper = mount(
  <Provider store={mockStore}>
    <VaultSelect />
  </Provider>
)

const noVaultWrapper = mount(
  <Provider store={noVaultsStore}>
    <VaultSelect />
  </Provider>
)

describe('VaultSelect component', () => {
  it('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders correctly if there are no vaults on state', () => {
    expect(toJson(noVaultWrapper)).toMatchSnapshot()
  })
})
