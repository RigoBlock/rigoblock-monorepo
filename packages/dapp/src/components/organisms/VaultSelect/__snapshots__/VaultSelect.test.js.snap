// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VaultSelect component renders correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [MockFunction],
      "getState": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(vaultSelect)>
    <vaultSelect
      dispatch={[MockFunction]}
      location="/vaults/0"
      vaults={
        Object {
          "0x421e1cef6e85e78da2470e54af64a626f45afb85": Object {
            "group": "0x7ce6e371085cb611fb46d5065397223ef2f952ff",
            "id": 2,
            "name": "Third Vault",
            "owner": "0x7328ef1d7ab7583eb9968b2f4a9c900f8a2e2d6d",
            "symbol": "das",
            "totalSupply": 20.23498,
          },
          "0x86a1ba4d485ce346bded508e2426798f825558be": Object {
            "group": "0x7ce6e371085cb611fb46d5065397223ef2f952ff",
            "id": 0,
            "name": "First Vault",
            "owner": "0x242b2dd21e7e1a2b2516d0a3a06b58e2d9bf9196",
            "symbol": "asd",
            "totalSupply": 14,
          },
        }
      }
    >
      <List
        Component={[Function]}
        className="vault-select"
        data={
          Array [
            Object {
              "className": "active",
              "id": 0,
              "itemName": "First Vault",
              "itemSymbol": "asd",
              "itemValue": 14,
              "onClick": [Function],
            },
            Object {
              "className": "",
              "id": 2,
              "itemName": "Third Vault",
              "itemSymbol": "das",
              "itemValue": 20.23498,
              "onClick": [Function],
            },
          ]
        }
      >
        <div
          className="vault-select"
        >
          <ListItem
            className="active"
            currencyGrowth={0}
            growth={0}
            id={0}
            itemName="First Vault"
            itemSymbol="asd"
            itemValue={14}
            key="0"
            onClick={[Function]}
          >
            <div
              className="list-item active"
              id={0}
              onClick={[Function]}
            >
              <ItemName
                name="First Vault"
                symbol="asd"
              >
                <div
                  className="no-events"
                >
                  <span
                    className="item-symbol"
                  >
                    asd
                  </span>
                  <span
                    className="item-name"
                  >
                    First Vault
                  </span>
                </div>
              </ItemName>
              <ItemValue
                currency="ETH"
                currencyGrowth={0}
                growth={0}
                itemValue={14}
                precision={2}
                valueSize="large"
              >
                <div
                  className="no-events"
                >
                  <span
                    className="item-value large"
                  >
                    14
                  </span>
                </div>
              </ItemValue>
            </div>
          </ListItem>
          <ListItem
            className=""
            currencyGrowth={0}
            growth={0}
            id={2}
            itemName="Third Vault"
            itemSymbol="das"
            itemValue={20.23498}
            key="1"
            onClick={[Function]}
          >
            <div
              className="list-item"
              id={2}
              onClick={[Function]}
            >
              <ItemName
                name="Third Vault"
                symbol="das"
              >
                <div
                  className="no-events"
                >
                  <span
                    className="item-symbol"
                  >
                    das
                  </span>
                  <span
                    className="item-name"
                  >
                    Third Vault
                  </span>
                </div>
              </ItemName>
              <ItemValue
                currency="ETH"
                currencyGrowth={0}
                growth={0}
                itemValue={20.23498}
                precision={2}
                valueSize="large"
              >
                <div
                  className="no-events"
                >
                  <span
                    className="item-value large"
                  >
                    20.23
                  </span>
                </div>
              </ItemValue>
            </div>
          </ListItem>
        </div>
      </List>
    </vaultSelect>
  </Connect(vaultSelect)>
</Provider>
`;

exports[`VaultSelect component renders correctly if there are no vaults on state 1`] = `
<Provider
  store={
    Object {
      "dispatch": [MockFunction],
      "getState": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(vaultSelect)>
    <vaultSelect
      dispatch={[MockFunction]}
      location="/vaults"
      vaults={Object {}}
    >
      <div
        className="vault-select"
      >
        Nothing here!
      </div>
    </vaultSelect>
  </Connect(vaultSelect)>
</Provider>
`;
